.github/workflows/build.yml

```
@@ -0,0 +1,998 @@
+name: Build
+on:
+  schedule:
+    # Run the workflow every day at 04:00 UTC
+    - cron: '0 4 * * *'
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  build:
+    outputs:
+      project-version: ${{ steps.version.outputs.project-version }}
+    runs-on: github-ubuntu-latest-m
+    name: Build Project
+    permissions:
+      id-token: write
+      contents: write
+    steps:
+      - uses: actions/checkout@08eba0b27e820071cde6df949e0beb9ba4906955 # v4.3.0
+      - uses: jdx/mise-action@c37c93293d6b742fc901e1406b8f764f6fb19dac # v2.4.4
+        with:
+          version: 2025.9.12
+      - name: Setup Maven Toolchains
+        shell: bash
+        run: .github/workflows/setup-maven-toolchains.sh
+      - uses: SonarSource/ci-github-actions/get-build-number@v1
+      - name: Setup DigiCert Client Tools
+        uses: SonarSource/ci-github-actions/code-signing@v1
+      - uses: SonarSource/ci-github-actions/config-maven@master # dogfood
+        env:
+          CURRENT_VERSION: skip
+          PROJECT_VERSION: skip
+        with:
+          common-mvn-flags: -Declipse.p2.mirrors=false -Dmaven.install.skip=true -Dmaven.test.skip=true -Dsonar.skip=true -Dcyclonedx.skip=false
+          artifactory-reader-role: private-reader
+      - name: Custom versioning
+        id: version
+        run: .cirrus/set_maven_build_version_github_action
+      - uses: SonarSource/ci-github-actions/build-maven@master # dogfood
+        with:
+          maven-args: -Declipse.p2.mirrors=false -Dmaven.install.skip=true -DskipTests -Dmaven.deploy.skip=true -Djacoco.append=true -Dsonar.coverage.jacoco.xmlReportPaths=${CIRRUS_WORKING_DIR}/org.sonarlint.eclipse.core.tests/target/site/jacoco-aggregate/jacoco.xml
+          artifactory-reader-role: private-reader
+          artifactory-deployer-role: qa-deployer
+          deploy-pull-request: true
+      - name: Upload site artifacts
+        if: always() && !cancelled()
```

```suggestion
        if: ${{ !cancelled() }}
```

We have similar situations in multiple places, the comment is general (not only for this occurrence)